File: 프로그래머스/0/120852. 소인수분해/소인수분해.py
def solution(n):
    answer = []
    start_num = 2
    
    while start_num <= n:
        if not n % start_num:
            if start_num not in answer:
                answer.append(start_num)
            n //= start_num
        else:
            start_num += 1
    return answer

File: 프로그래머스/0/120852. 소인수분해/소인수분해.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public List<Integer> solution(int n) {
        List<Integer> answer = new ArrayList<>();
        int startNum = 2;
        
        while (startNum <= n) {
            if (n % startNum == 0) {
                if (!answer.contains(startNum)) {
                    answer.add(startNum);
                }
                
                n /= startNum;
            } else {
                startNum++;
            }
        }
        return answer;
    }
}